tests/validate/bandwidth_test/src/host.cpp

```diff
@@ -151,101 +161,184 @@ int main(int argc, char** argv) {
+            OCL_CHECK(err, output_buffer = cl::Buffer(context, CL_MEM_READ_WRITE, vector_size_bytes, nullptr, &err));
+
+            OCL_CHECK(err, err = krnls[NUM_KERNEL - 1].setArg(0, input_buffer));
+            OCL_CHECK(err, err = krnls[NUM_KERNEL - 1].setArg(1, output_buffer));
+            OCL_CHECK(err, err = krnls[NUM_KERNEL - 1].setArg(2, DATA_SIZE));
+            OCL_CHECK(err, err = krnls[NUM_KERNEL - 1].setArg(3, reps));
+
+            OCL_CHECK(err, err = q.enqueueWriteBuffer(input_buffer, CL_TRUE, 0, vector_size_bytes, input_host.data(),
                                                       nullptr, nullptr));
             OCL_CHECK(err, err = q.finish());
-        }
 
-        std::chrono::high_resolution_clock::time_point timeStart;
-        std::chrono::high_resolution_clock::time_point timeEnd;
+            std::chrono::high_resolution_clock::time_point timeStart;
+            std::chrono::high_resolution_clock::time_point timeEnd;
 
-        timeStart = std::chrono::high_resolution_clock::now();
+            timeStart = std::chrono::high_resolution_clock::now();
 
-        for (int i = 0; i < NUM_KERNEL; i++) {
-            OCL_CHECK(err, err = q.enqueueTask(krnls[i]));
-        }
-        q.finish();
-        timeEnd = std::chrono::high_resolution_clock::now();
+            OCL_CHECK(err, err = q.enqueueTask(krnls[NUM_KERNEL - 1]));
+            q.finish();
+            timeEnd = std::chrono::high_resolution_clock::now();
 
-        for (int i = 0; i < NUM_KERNEL; i++) {
-            OCL_CHECK(err, err = q.enqueueReadBuffer(output_buffer[i], CL_TRUE, 0, vector_size_bytes,
-                                                     output_host[i].data(), nullptr, nullptr));
+            OCL_CHECK(err, err = q.enqueueReadBuffer(output_buffer, CL_TRUE, 0, vector_size_bytes, output_host.data(),
+                                                     nullptr, nullptr));
             OCL_CHECK(err, err = q.finish());
-        }
 
-        // check
-        for (int i = 0; i < NUM_KERNEL; i++) {
+            // check
             for (uint32_t j = 0; j < DATA_SIZE; j++) {
-                if (output_host[i][j] != input_host[j]) {
-                    printf("ERROR : kernel failed to copy entry %i input %i output %i\n", j, input_host[j],
-                           output_host[i][j]);
+                if (output_host[j] != input_host[j]) {
+                    std::cout << "ERROR : kernel failed to copy entry " << j << " input " << input_host[j] << " output "
+                              << output_host[j] << std::endl;
                     return EXIT_FAILURE;
                 }
             }
-        }
 
-        double usduration;
-        double dnsduration;
-        double dsduration;
-        double bpersec;
-        double mbpersec;
+            double usduration;
+            double dnsduration;
+            double dsduration;
+            double bpersec;
+            double mbpersec;
```

Comment:
Combine with usage.    It all looks like deja vu, I didn't check, but is refactoring possible?